### DISCUSSION TITLE
Is it possible to mix CUDA Fortran and SYCL?

### DISCUSSION THREAD

**Discussion by AGindinson** [intel/llvm#3952] [General]
Reposting @Ruyk's concern from https://github.com/intel/llvm/issues/3947 here:

"I am working on an example (codeplaysoftware/SYCL-For-CUDA-Examples#16) where I mix CUDA Fortran with SYCL code. Because nvfortran does its own magic at linking, I cannot use clang for linking the final binary. The only way I could put together the two is if I create a library with the SYCL code and link that with the Fortran binary.
@AGindinson is there any way to trigger the "sycl-post-link processing/SPIR-V Translator/AOT compilation" without creating a library or an executable?"

**Comment by AGindinson** [comment]
@Ruyk, I'm not sure if I understand how the two aspects fit together:

> The only way I could put together the two is if I create a library with the SYCL code

> Is there any way to trigger the "sycl-post-link processing/SPIR-V Translator/AOT compilation" without creating a library?

Is the second point regarding the SYCL side of the application? If so, why is library creation not an option?

**Reply by Ruyk** [reply]
The main thing I encountered so far is that it complicates integration with build systems. I am using a very trivial makefile and example there, but imagine a project that combines CUDA fortran and SYCL CUDA files. If I have to modify all cmake or configure scripts to generate the SYCL files as libraries, it is a noticeable burden only needed for the SYCL backend.
